tests/bson/bson-toPHP-015.phpt

```diff
@@ -0,0 +1,140 @@
+--TEST--
+MongoDB\BSON\toPHP(): Assigning a type map value other than default disables persistable behaviour
```

Comment:
I added this test to ensure adding the `bson` type map type doesn't accidentally break existing behaviour.

Reply:
This line is not entirely true. Per https://www.php.net/manual/en/mongodb.persistence.deserialization.php, a `__pclass` property takes priority over a class name iff the `__pclass` exists and implements Persistable.

All _other_ type map values (i.e. "stdClass", "object", "array", and now "bson") supersede the `__pclass`.

Reply:
I appreciate the additional tests, but I don't think my original comment was addressed. This sentence is not true:

> Assigning a type map value other than default disables persistable behaviour

Persistable behavior may _not_ be disabled if the type map value is the FQCN of a class implementing Unserializable. I'd propose renaming this test to:

> Type map values can override Persistable behavior

And then optionally leave a comment within the test like so:

> The special type map values below always override Persistable behavior. Note that an Unserializable class name will only override Persistable behavior if the class specified in __pclass does not exist or does not implement Persistable.

Reply:
Updated description, created PHPC-2178 to track additional tests for `toPHP()`.